control-plane/agents/src/bin/ha/cluster/switchover.rs

```diff
@@ -243,48 +247,55 @@ impl SwitchOverEngine {
-                            Stage::RepublishVolume => q.republish_volume(&etcd).await,
-                            Stage::PublishPath => q.publish_path(&etcd, &nodes).await,
-                            Stage::DeleteTarget => q.delete_target(&etcd).await,
-                            Stage::Successful => match q.delete_switchover(&etcd).await {
-                                Ok(_) => break,
-                                Err(e) => Err(e),
-                            },
-                            _ => break,
-                        };
-                        match result {
-                            Ok(_) => {
-                                // reset retry delay back to the start.
-                                retry_delay = retry_delay_min;
-                            }
-                            Err(_) => {
-                                retry_delay = (retry_delay * 3).min(retry_delay_max);
-                                tokio::time::sleep(retry_delay).await;
-                            }
+        loop {
+            if let Some(mut q) = recv.lock().await.recv().await {
+                let retry_delay_min = std::time::Duration::from_secs(1);
+                let retry_delay_max = std::time::Duration::from_secs(384);
+                let mut retry_delay = retry_delay_min;
+                loop {
+                    // TODO: handle error handling properly
+                    let result = match q.stage {
+                        Stage::Init => q.initialize(&etcd).await,
+                        Stage::RepublishVolume => q.republish_volume(&etcd).await,
+                        Stage::PublishPath => q.publish_path(&etcd, &nodes).await,
+                        Stage::DeleteTarget => q.delete_target(&etcd).await,
+                        Stage::Successful => match q.delete_switchover(&etcd).await {
+                            Ok(_) => break,
+                            Err(e) => Err(e),
+                        },
+                        _ => break,
+                    };
+                    match result {
+                        Ok(_) => {
+                            // reset retry delay back to the start.
+                            retry_delay = retry_delay_min;
+                        }
+                        Err(_) => {
+                            retry_delay = (retry_delay * 3).min(retry_delay_max);
+                            tokio::time::sleep(retry_delay).await;
                         }
                     }
                 }
             }
-        });
+        }
+    }
+
+    /// Mover instantiates worker task to asynchronously process Switchover request.
+    pub fn mover(&self, recv: Arc<Mutex<UnboundedReceiver<SwitchOverRequest>>>) {
+        for i in 0 .. WORKER_NUM {
+            info!("Starting worker-{} of Switchover Engine", i.to_string());
+            let cloned_self = self.clone();
+            let cloned_channel = Arc::clone(&recv);
```

Comment:
Clone takes ref self so you can just call clone:
```suggestion
            let cloned_channel = recv.clone();
```

Reply:
Incorporated